src/app/swap/swap-listener.ts

```diff
@@ -0,0 +1,68 @@
+import { toSats } from "@domain/bitcoin"
+
+import { admin as LedgerAdmin } from "@services/ledger/admin"
+import { WalletCurrency } from "@domain/shared"
+import { LedgerTransactionType } from "@domain/ledger"
+import { SwapProvider, SwapState, SwapType } from "@domain/swap"
+import { addAttributesToCurrentSpan } from "@services/tracing"
+import { SwapErrorNoActiveLoopdNode } from "@domain/swap/errors"
+import { sha256 } from "@domain/bitcoin/lightning"
+
+export const startSwapMonitor = async (swapService: ISwapService) => {
+  const isSwapServerUp = await swapService.healthCheck()
+  if (isSwapServerUp instanceof Error) return new SwapErrorNoActiveLoopdNode()
+  const listener = swapService.swapListener()
+  listener.on("data", (response) => {
+    handleSwapOutCompleted(response)
+  })
+}
+
+export const handleSwapOutCompleted = async (swapStatus: SwapStatusResultWrapper) => {
+  if (swapStatus.parsedSwapData) {
+    const type = swapStatus.parsedSwapData.swapType
+    const onchainMinerFee = toSats(swapStatus.parsedSwapData.onchainMinerFee)
+    const offchainRoutingFee = toSats(swapStatus.parsedSwapData.offchainRoutingFee)
+    const serviceProviderFee = toSats(swapStatus.parsedSwapData.serviceProviderFee)
+    const totalFees = onchainMinerFee + offchainRoutingFee + serviceProviderFee
+    const state = swapStatus.parsedSwapData.state
+    if (type === SwapType.Swapout && state === SwapState.Success && totalFees > 0) {
+      const swapId = swapStatus.parsedSwapData.id as SwapId
+      const swapFeeMetadata: SwapTransactionMetadataUpdate = {
```

Comment:
Creating the XxxMetadata` types should be delegated to the `LedgerFacade` where they should be unit tested. See the functions [here](https://github.com/GaloyMoney/galoy/blob/main/src/services/ledger/tx-metadata.ts) exposed in the facade [here](https://github.com/GaloyMoney/galoy/blob/main/src/services/ledger/facade.ts#L19) and tested [here](https://github.com/GaloyMoney/galoy/blob/main/test/unit/services/ledger/domain/tx-metadata.spec.ts).